ReactAndroid/src/main/java/com/facebook/react/modules/statusbar/StatusBarModule.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ * <p/>
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+
+package com.facebook.react.modules.statusbar;
+
+import android.animation.ArgbEvaluator;
+import android.animation.ValueAnimator;
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.os.Build;
+import android.support.v4.view.ViewCompat;
+import android.view.View;
+
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.UiThreadUtil;
+
+public class StatusBarModule extends ReactContextBaseJavaModule {
+
+  private static final String ERROR_NO_ACTIVITY = "Tried to change the status bar while not attached to an Activity";
+
+  private int mWindowFlags = 0;
+
+  public StatusBarModule(ReactApplicationContext reactContext) {
+    super(reactContext);
+  }
+
+  @Override
+  public String getName() {
+    return "StatusBarAndroid";
```

Comment:
Why not call this RCTStatusBarManager? then you can avoid the forking in the JS code.